doc/user-manual/language/cubical.lagda.rst

```diff
@@ -1,106 +1,930 @@
+a cubical identity type. The
+https://github.com/agda/cubical/blob/master/Cubical/Core/Id.agda file
+exports all of the primitives for this type, including the notation
+``_≡_`` and a ``J`` eliminator that computes definitionally on
+``refl``.
+
+The cubical identity type and the path type are equivalent, so all of
+the results for one can be transported to the other one (using
+univalence). Using this we have implemented an interface to HoTT/UF in
+https://github.com/agda/cubical/blob/master/Cubical/Core/HoTT-UF.agda
+which provides the user with the key primitives of Homotopy Type
+Theory and Univalent Foundations implemented using cubical primitives
+under the hood. This hence gives an axiom free version of HoTT/UF
+which computes properly.
+
+.. code-block:: agda
+
+  module Cubical.Core.HoTT-UF where
+
+  open import Cubical.Core.Id public
+     using ( _≡_            -- The identity type.
+           ; refl            -- Unfortunately, pattern matching on refl is not available.
+           ; J              -- Until it is, you have to use the induction principle J.
+
+           ; transport      -- As in the HoTT Book.
+           ; ap
+           ; _∙_
+           ; _⁻¹
+
+           ; _≡⟨_⟩_         -- Standard equational reasoning.
+           ; _∎
+
+           ; funExt         -- Function extensionality
+                            -- (can also be derived from univalence).
+
+           ; Σ              -- Sum type. Needed to define contractible types, equivalences
+           ; _,_            -- and univalence.
+           ; pr₁            -- The eta rule is available.
+           ; pr₂
+
+           ; isProp         -- The usual notions of proposition, contractible type, set.
+           ; isContr
+           ; isSet
+
+           ; isEquiv        -- A map with contractible fibers
+                            -- (Voevodsky's version of the notion).
+           ; _≃_            -- The type of equivalences between two given types.
+           ; EquivContr     -- A formulation of univalence.
+
+           ; ∥_∥             -- Propositional truncation.
+           ; ∣_∣             -- Map into the propositional truncation.
+           ; ∥∥-isProp       -- A truncated type is a proposition.
+           ; ∥∥-recursion    -- Non-dependent elimination.
+           ; ∥∥-induction    -- Dependent elimination.
+           )
+
+In order to get access to only the HoTT/UF primitives start a file as
+follows:
+
+.. code-block:: agda
+
+  {-# OPTIONS --cubical #-}
+
+  open import Cubical.Core.HoTT-UF
+
+However, even though this interface exists it is still recommended
+that one uses the cubical identity types unless one really need ``J``
+to compute on ``refl``. The reason for this is that the syntax for
+path types does not work for the identity types, making many proofs
+more involved as the only way to reason about them is using ``J``
+(furthermore, the path types satisfy many useful definitional
+equalities that the identity types don't).
+
```

Comment:
The parenthetical remark could be made into a sentence of its own. It’s actually a very valid reason for avoid identity types, maybe more so than the syntax.

Reply:
Good point! Done